changes/2020-06-04_how-to-fail-with-keyrings/change.md

```diff
@@ -0,0 +1,216 @@
+will affect implementations.
+
+## Definitions
+
+### Conventions used in this document
+
+The key words
+"MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+"SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL"
+in this document are to be interpreted as described in
+[RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+## Summary
+
+In order to define when keyrings MUST fail,
+we define a new concept of the "decryption contract",
+how it applies to keyrings generally,
+and how it is used to determine when keyrings MUST fail.
+
+## Out of Scope
+
+- The specifics of how ESDK components
+  communicate failure information between themselves
+  is out of scope.
+  We will address this in future specification changes.
+
+- The specific nature of _how_ ESDK framework components indicate failure
+  is out of scope.
+  That will depend on the best practices for each language.
+
+- Defining the decryption contract for each keyring
+  is out of scope.
+  We will address this in future specification changes.
+
+- A mechanism to interrogate a keyring
+  to determine the properties of its decryption contract
+  is out of scope.
+  We SHOULD address this in future changes.
+
+- This change only applies to keyrings.
+  We SHOULD define analogous concepts for
+  cryptographic materials managers
+  and the AWS Encryption SDK client
+  in future changes.
+
+## Motivation
+
+We lack a consistent definition both of
+what it means for a keyring to fail
+and what causes a keyring to fail.
+Without this consistent definition,
+different keyrings and implementations
+do slightly different things,
+leading to implementation drift
+and confusion for users and implementors.
+
+In order to clarify this situation,
+we MUST first define what it means for a keyring to fail,
+and that means defining what it means for a keyring to succeed.
+To do this,
+we define a new concept of the
+"keyring decryption contract".
+
+## Drawbacks
+
+This change SHOULD not introduce any drawbacks.
+It is simplifying and clarifying
+a relationship that already exists
+in the specification
+and clearly defining
+how that relationship affects behavior.
+
+## Security Implications
+
+The keyring decryption contract
+simplifies the definition of
+the security properties of a keyring
+and provides a consistent way to describe
+the security properties of each keyring.
+
+## Operational Implications
+
+Clearly defining what causes keyrings to fail
+makes it simpler for users to predict failing behavior
+based on either environment
+and across implementations.
+
+## Guide-level Explanation
+
+When a keyring succeeds on encryption,
+it creates an artifact: one or more encrypted data keys.
+This artifact has certain requirements that
+MUST be met on decryption
+in order to obtain the plaintext data key.
+This set of requirements is what we call
+the "keyring decryption contract."
+The process that a keyring uses to create this artifact
+is largely irrelevant to the decryption contract.
```

Comment:
```suggestion
is only relevant as a mechanism to fulfill the decryption contract.
```

Or something positive.  The keyring is responsible for fulfilling the contract, and if the artifact can help enforce that great, otherwise, as you say it is not relevant.